ttyplot.c

```diff
@@ -301,8 +301,14 @@ int main(int argc, char *argv[]) {
         exit(0);
     }
 
-    // Reset the getopt index as per spec.
-    optind = 0;
+    // Run a 2nd iteration over the arguments to actually process the options.
+    // According to getopt's documentation this is done by setting optind to 1
+    // (or 0 in some special cases). On BSDs and Macs optreset must be set to 1
+    // in addition.
+    optind = 1;
+#if defined(__OpenBSD__) || (defined(__APPLE__) && defined(__MACH__))
+    optreset = 1;
+#endif
```

Comment:
- What's the idea about `__MACH__` here?
- Are we sure yet that (or when) `optind` 1 is fine for glibc?
- Here's how dnsdb solves the same problem differently: https://github.com/dnsdb/dnsdbq/blob/a3b77edda4b4f66d27e8b12c2c551c8dd1c2dcd4/dnsdbq.c#L1181-L1196

Reply:
@MIvanchev yeah could you add ifs for glibc/nonglibc, I like how dnsdbq does it

we should test on some non-glibc linux as well

Reply:
PS: This is where glibc's getopt is doing additonal work for `optind` 0: https://github.com/bminor/glibc/blob/b7f8b6b64b135e6dba5083d688675874d6809c91/posix/getopt.c#L482

Reply:
> we should test on some non-glibc linux as well

Alpine has musl, seems like a good candidate

Reply:
I do not find **any** indication that glibc mandates 0 vs 1 so I believe dnsdbq is cargo culting in this case. However it might make sense to check for `__FreeBSD__` / `__NetBSD__`.

Reply:
@hartwork 
> What's the idea about __MACH__ here?
IMHO __APPLE__ and __MACH__ together indicate a MacOS installation. I don't think we need to check further than that but I'm not a Mac user so open for suggestions.

> Are we sure yet that (or when) optind 1 is fine for glibc?
Yes.

Reply:
> I do not find **any** indication that glibc mandates 0 vs 1 so I believe dnsdbq is cargo culting in this case.

@MIvanchev did you see my link to glibc code above?:

> PS: This is where glibc's getopt is doing additonal work for `optind` 0: https://github.com/bminor/glibc/blob/b7f8b6b64b135e6dba5083d688675874d6809c91/posix/getopt.c#L482

Reply:
Yes I did, it does additional work for the 0 case but we do not require the additional work as the documentation states. No extensions are used and `POSIXLY_CORRECT` is not changed in-between.

Reply:
> IMHO **APPLE** and **MACH** together indicate a MacOS installation.

@MIvanchev if https://sourceforge.net/p/predef/wiki/OperatingSystems/#macos is correct — and they usually are — then:
- It's macOS X specifically, with emphasis on X.
- Only set by GNU C and the Intel compiler.

Unless we have strong need for `__MACH__` that I don#t yet see, I'd vote for dropping `__MACH__`.

Actually I think I'd vote to exact-copy dnsdb's solution. PS: I read that MirBSD seems to also define the one for OpenBSD so it should be covered for free.

Reply:
> Yes I did, it does additional work for the 0 case but we do not require the additional work as the documentation states. No extensions are used and `POSIXLY_CORRECT` is not changed in-between.

@MIvanchev cool, thanks! :+1:

Reply:
I dropped `__MACH__` but I'm leaving `optind = 1`. dnsdb is really just paranoid here for no reason at all.

Reply:
@MIvanchev their approach is arguably more robust and can be copy pasted for re-use elsewhere even to places where extensions are used, and they code will survive if someone adds an extension without noticing. Maybe they don't need it today, but paranoid being negative in connotation don't see them doing anything wrong, rather the contrary. Just my 2 cents.

Reply:
You're really overthinking this IMO, just because someone did something somewhere doesn't mean it's a good idea. Cargo cult or false interpretation of the documentation is not robust programming. This `__MACH__` stuff was a different beast and really increases robustness.

Regarding your other question about the well-defined order for `-v` and `-h`. If you have:

```
ttyplot -v -h
ttyplot -h -v
ttyplot -v -v -v -h
```

One should always have a fixed precedence over the other.

Reply:
> You're really overthinking this IMO, just because someone did something somewhere doesn't mean it's a good idea.

@MIvanchev I don't need to be right about anything here, I just like to understand things better and be sure we fix a good way. I'm not sure where you read me say anything like that, it doesn't sound like what I intended to say or said literally.

> Cargo cult or false interpretation of the documentation is not robust programming.

Not sure how that relates.

> The code will survive anyhow because the second you add extensions it won't work without `= 0` and you'll just change it. 

Not work somewhere != not work everywhere consistently. It may go unnoticed in a project with Linux-only CI for instance, e.g. ttyplot.

> This `__MACH__` stuff was a different beast and really increases robustness.

I don't see how. But before either of us has access to actual macOS 9 we're probably both guessing.

> Regarding your other question about the well-defined order for `-v` and `-h`. If you have:
> 
> ```
> ttyplot -v -h
> ttyplot -h -v
> ttyplot -v -v -v -h
> ```
> 
> One should always have a fixed precedence over the other.

How/why is a fixed preference more important than order with help and version?

Reply:
It was about the dnsdbq crew, not you, I understand and value your motivation :)

>How/why is a fixed preference more important than order with help and version?

I mean that the program does the same with `-v -h` as with `-h -v` and the order doesn't matter.